src/wallet/wallet.cpp

```diff
@@ -3936,9 +3936,11 @@ CKeyPool::CKeyPool(const CPubKey& vchPubKeyIn, bool internalIn)
 
 int CWalletTx::GetDepthInMainChain(interfaces::Chain::Lock& locked_chain) const
 {
+    assert(pwallet != nullptr);
+    AssertLockHeld(pwallet->cs_wallet);
     if (isUnconfirmed() || isAbandoned()) return 0;
 
-    return locked_chain.getBlockDepth(m_confirm.hashBlock) * (isConflicted() ? -1 : 1);
+    return (pwallet->GetLastBlockHeight() - m_confirm.block_height + 1) * (isConflicted() ? -1 : 1);
```

Comment:
In commit "Use CWallet::m_last_block_processed_height in GetDepthInMainChain" (0ff03871add000f8b4d8f82aeb168eed2fc9dc5f)

@ariard It seems like this line is written assuming that `GetLastBlockHeight()` is always `>= m_confirm.block_height`, but if I add asserts here:

```c++
assert(m_confirm.block_height >= 0);
assert(m_confirm.block_height <= pwallet->GetLastBlockHeight());
```

The second assert fails in `wallet_abandonconflict.py` and `wallet_bumpfee.py` tests. Wondering if this expected or seems like a bug to you